Many changes in the markdown documentation preparing for v1.0

cristian-recoseanu opened the pull request:
Some of the changes:
- Changed the advertised DNS-SD service type name
- Removed mentions of the AMQP protocol
- Made both WebSocket and MQTT mandatory
- Refactoring of the IS-05 transport parameters which are now 3 instead of 2
- More emphasis that object event type is out of scope for 1.0
- Refactored REST API (Late joiners) to Event & Tally REST API
- Adds measurement units guidelines section
- New section called 2.0. Message_types (other documents changed prefix index)
- Mention of health responses in the WebSocket section
- Adds more clarity in certain sections

cristian-recoseanu committed:
Changed the advertised DNS-SD service type for the MQTT broker

Changed the advertised DNS-SD service type for the MQTT broker from _nmos-events_broker-mqtt to _nmos-mqtt.
This was done because the maximum characters of a DNS-SD service type name is 15.

cristian-recoseanu committed:
Removed any mentions of the AMQP protocol

Removed any mentions of the AMQP protocol because of feedback received from the community which highlighted that having too many protocols may create incompatible offerings across multiple vendors and create confusion

cristian-recoseanu committed:
Clarification that both transports are mandatory capabilities

Added clarification that both transports (WebSocket and MQTT) are mandatory capabilities.

cristian-recoseanu committed:
Renamed the IS-05 parameters and added a distinct parameter for the websocket url

Renamed the IS-05 parameters and added a distinct parameter for the websocket url.
event_address is now two parameters broker_topic and websocket_url.
event_rest_url is now rest_api_url.

This change was made in order to keep the IS-05 transport parameters agnostic of their use and generic/reusable

cristian-recoseanu committed:
Adding statements which emphasise that object event_types are out of scope for v1.0

Adding statements which emphasise that object event_types are out of scope for v1.0

cristian-recoseanu committed:
Refactored REST API (Late joiners) to Event & Tally REST API

Refactored REST API (Late joiners) to Event & Tally REST API because the API now covers more than just late joiners sate synchronisation but also metadata surrounding the event_type emitted using the /type path

cristian-recoseanu committed:
Adds clarifications about url trailing slashes and url template

Adds clarifications about url trailing slashes for the urn:x-nmos:events:query/v1.0 control href and the rest_api_url transport parameter url.
Adds clarifications about the template a system controller should use when creating the rest_api_url parameter and how a receiver is expected to use this.

cristian-recoseanu committed:
Clairifcations around mandatory or optional fields and usage of wildcard

cristian-recoseanu committed:
Added Measurement units guidelines section and made references to it in relevant sections

cristian-recoseanu committed:
Changed the websocket health value field to timestamp

cristian-recoseanu committed:
Introduced a new document called 2.0. Message_types

Introduced a new document called 2.0. Message_types with the following consequences :
- some documents have had their prefix number pushed by 1
- had to fix some links caused by some documents having their prefix number being pushed by 1

The Message_types document is WIP (placeholder)
Made a reference to it from Core_models in the message structure section

cristian-recoseanu committed:
Removed message structure from Core_models (to be moved in Message_types)

cristian-recoseanu committed:
Changes health time from 60 to 10 seconds and adds a health response example

cristian-recoseanu committed:
Adds content to the Message_types placeholder

- Introduction
- the state message type
- the reboot message type
- the shutdown message type
- the connection lost message type
- the health message type

datkins47 commented:
Should we extend this to cover m,g,t,p and the equivalent bytes version

 


### 2.3 Prefixes

 	
			
		
The use of prefixes is allowed but discouraged apart from SI base units (e.g. `kg`).

		
The following further remarks apply to prefixes:

			
		
* micro (10^6) will be prefixed with `u`

		
* `k` prefix will always stand for 1000 units and not 2^10 (1024); for the latter use `kB`

 

`m` prefix will always stand for 1000,000 units and not 2^20 ; for the latter use `kB`

`g` prefix will always stand for 1000,000,000 units and not 2^30 ; for the latter use `mB`

`t` prefix will always stand for 1000,000,000,000 units and not 2^40 ; for the latter use `tB`

`p` prefix will always stand for 1000,000,000,000,000 units and not 2^50 ; for the latter use `pB`

 

But is not the case that generally the use is

 

Kb is kilo bits  1000 bits

 

kB is kilo Bytes (now not sure )

 

we have always been internally consistent and used 1000’s because they are human and when you get to G, T etc are far less confusing…..

 

 

 

David

 

From: cristian-recoseanu [mailto:notifications@github.com] 
Sent: Friday, October 19, 2018 5:14 PM
To: AMWA-TV/nmos-event-tally
Cc: David Atkins; Review requested
Subject: Re: [AMWA-TV/nmos-event-tally] Many changes in the markdown documentation preparing for v1.0 (#13)

 

@cristian-recoseanu <https://github.com/cristian-recoseanu>  requested your review on: #13 <https://github.com/AMWA-TV/nmos-event-tally/pull/13>  Many changes in the markdown documentation preparing for v1.0.

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub <https://github.com/AMWA-TV/nmos-event-tally/pull/13#event-1915169003> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AO7cG2iKyfj-w8RLY8Tr1mXoySGdSkdxks5umfovgaJpZM4XwqYq> .  <https://github.com/notifications/beacon/AO7cG-fOzZIwoOoL_dVc9NYthmHh2g5Gks5umfovgaJpZM4XwqYq.gif>

datkins47 commented:
Should we set health timeout to be 10 seconds and your dead, and recommend a heartbeat every 5 seconds

 

Can’t find the reference on the is-04 docs at present!

 

David

 

From: cristian-recoseanu [mailto:notifications@github.com] 
Sent: Friday, October 19, 2018 5:14 PM
To: AMWA-TV/nmos-event-tally
Cc: David Atkins; Review requested
Subject: Re: [AMWA-TV/nmos-event-tally] Many changes in the markdown documentation preparing for v1.0 (#13)

 

@cristian-recoseanu <https://github.com/cristian-recoseanu>  requested your review on: #13 <https://github.com/AMWA-TV/nmos-event-tally/pull/13>  Many changes in the markdown documentation preparing for v1.0.

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub <https://github.com/AMWA-TV/nmos-event-tally/pull/13#event-1915169003> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AO7cG2iKyfj-w8RLY8Tr1mXoySGdSkdxks5umfovgaJpZM4XwqYq> .  <https://github.com/notifications/beacon/AO7cG-fOzZIwoOoL_dVc9NYthmHh2g5Gks5umfovgaJpZM4XwqYq.gif>

datkins47 commented:
sent comments by email.....

peterbrightwell commented:
From the IS-04 docs (http://amwa-tv.github.io/nmos-discovery-registration/tags/v1.2_spec/docs/4.1._Behaviour_-_Registration.html):

Nodes are expected to perform a heartbeat every 5 seconds by default.

Registration APIs should use a garbage collection interval of 12 seconds by default (triggered just after two failed heartbeats at the default 5 second interval).

It is recommended that heartbeat and garbage collection intervals are user-configurable to non-default values in Nodes and Registration APIs respectively. The garbage collection interval may be increased without any adverse effects on the registering Nodes. Decreasing the garbage collection interval, or increasing the heartbeat interval requires a system-wide configuration.

Nodes are only required to perform a heartbeat to maintain their ‘Node’ resource in the Registration API. If heartbeats fail over a period greater than the garbage collection interval, both the Node and all registered sub-resources should be removed from the registry automatically.

(Also, the full-stack team are considering a system-wide setting for the required heartbeat interval, conveyed by the "System Resource").

mjeras commented on docs/5.1. Transport - MQTT.md:
```diff
@@ -0,0 +1,77 @@
+
+MQTT publishers are required to send the `retained message` flag with every message in order to solve the problem of late joiners.
+
+Consumers (receivers) have a choice to either use the retained message or query the late joiners api using `rest_api_url` to access the latest state of an emitter in order to get in sync.
```
Maybe we should mention the will message here and reference to the messages document?

cristian-recoseanu replied:
Made the necessary changes as per conversations via email (github comments were not working)

mjeras commented on docs/5.2. Transport - Websocket.md:
```diff
@@ -108,14 +119,26 @@ A subscription command will only get sent out as a consequence of an NMOS Connec
 ### 4.1 Heartbeats
 
-Upon connection, the client is required to report its health every 60 seconds in order to maintain its session and subscription. This is similar to the behaviour required by the registry for nodes in IS-04 specification.
+Upon connection, the client is required to report its health every 10 seconds in order to maintain its session and subscription. This is similar to the behaviour required by the registry for nodes in IS-04 specification. Every `health` command should be followed by a `health` response (see [Message types](2.0.%20Message_types.md)).
```
As Peter mentioned in his comment, 5 is the default for IS-04 (but can be changed). Should we implement the same defaults?

mjeras commented on docs/5.1. Transport - MQTT.md:
```diff
@@ -0,0 +1,77 @@
+
+MQTT publishers are required to send the `retained message` flag with every message in order to solve the problem of late joiners.
+
+Consumers (receivers) have a choice to either use the retained message or query the late joiners api using `rest_api_url` to access the latest state of an emitter in order to get in sync.
```
Maybe we should mention the will message here and add a reference to the messages document?

cristian-recoseanu replied:
Made the necessary changes as per conversations via email (github comments were not working)

mjeras commented on docs/7.0. Measurement_units_guidelines.md:
```diff
@@ -0,0 +1,32 @@
+
+### 2.1. Case sensitive behaviour
+
+Units of measurement can be case sensitive and this behaviour is allowed.
```
I think we should say MUST instead of "can", as otherwise you cannot distinguish some values (e.g. mega and milli; Siemens and second etc.)

mjeras commented on docs/7.0. Measurement_units_guidelines.md:
```diff
@@ -0,0 +1,32 @@
+
+### 2.3 Prefixes
+
+The use of prefixes is allowed but discouraged apart from SI base units (e.g. `kg`).
```
I don't think we should say "e.g." here as "kg" is the only base SI unit that is prefixed. "dB" is another case, but that one is never prefixed with anything else so we don't need to mention it

mjeras commented on docs/7.0. Measurement_units_guidelines.md:
```diff
@@ -0,0 +1,32 @@
+The use of prefixes is allowed but discouraged apart from SI base units (e.g. `kg`).
+The following further remarks apply to prefixes:
+
+* micro (10^6) will be prefixed with `u`
```
micro is 10^-6, not 10^6

mjeras commented on docs/7.0. Measurement_units_guidelines.md:
```diff
@@ -0,0 +1,32 @@
+The following further remarks apply to prefixes:
+
+* micro (10^6) will be prefixed with `u`
+* `k` prefix will always stand for 1000 units and not 2^10 (1024); for the latter use `kB`
```
I would mandate that k, M, G etc. are always power of ten, even for bytes and bits as using one notation is the only way to achieve consistency.
Also, we should add M and G and they maybe just say etc.

cristian-recoseanu commented:
Hi David,

In reply to your measurements comments.
I agree with having more of the prefix cases covered.

Now looking at the symbols actually I think for 1024 (2^10) it's actually **KiB (kibibyte)** for 1024 bytes.

Now I'm wondering if we should use ISO/IEC 80000 standard for all of these.

[https://en.wikipedia.org/wiki/Kibibyte](https://en.wikipedia.org/wiki/Kibibyte)
[https://en.wikipedia.org/wiki/Kibibit](https://en.wikipedia.org/wiki/Kibibit)

cristian-recoseanu committed:
Corrected some mistakes and added some more examples for prefixes

Made clear that units have to be case sensitive
Fixed the micro 10^-6 where the minus was missed
Added more examples for M and G prefixes

cristian-recoseanu committed:
Adds a section on MQTT WILL message which briefly explains WILL messages and links to the Message types document

cristian-recoseanu committed:
Changed websocket heartbeat time to 5s and added some clarity around expectations

peterbrightwell commented:
Just noticing the discussion on prefixes... that looks like something that should be kept consistent across NMOS specifications. I will comment further on Basecamp.

peterbrightwell commented:
Just noticing the discussion on prefixes... that looks like something that should be kept consistent across NMOS specifications. I will comment further on Basecamp.

peterbrightwell commented:
Just noticing the discussion on prefixes... that looks like something that should be kept consistent across NMOS specifications. I will comment further on Basecamp.

P

From: cristian-recoseanu <notifications@github.com>
Reply-To: AMWA-TV/nmos-event-tally <reply@reply.github.com>
Date: Monday, 22 October 2018 at 10:34
To: AMWA-TV/nmos-event-tally <nmos-event-tally@noreply.github.com>
Cc: Peter Brightwell <peter.brightwell@bbc.co.uk>, Comment <comment@noreply.github.com>
Subject: Re: [AMWA-TV/nmos-event-tally] Many changes in the markdown documentation preparing for v1.0 (#13)


Hi David,

In reply to your measurements comments.
I agree with having more of the prefix cases covered.

Now looking at the symbols actually I think for 1024 (2^10) it's actually KiB (kibibyte) for 1024 bytes.

Now I'm wondering if we should use ISO/IEC 80000 standard for all of these.

https://en.wikipedia.org/wiki/Kibibyte
https://en.wikipedia.org/wiki/Kibibit

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/AMWA-TV/nmos-event-tally/pull/13#issuecomment-431777932>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AOvCyNYomwKgDCOooEXGaoLf_b6M3Tshks5unZEfgaJpZM4XwqYq>.



----------------------------

http://www.bbc.co.uk
This e-mail (and any attachments) is confidential and may contain personal views which are not the views of the BBC unless specifically stated.
If you have received it in error, please delete it from your system.
Do not use, copy or disclose the information in any way nor act in reliance on it and notify the sender immediately.
Please note that the BBC monitors e-mails sent or received.
Further communication will signify your consent to this.

---------------------

cristian-recoseanu commented:
@datkins47 are you happy with the latest changes I made to measurements and heartbeats?
I think we should stay clear of specifying too much in the IS-07 measurements guidelines as we may find it is a grander activity across NMOS (Peter made a Basecamp post about this).
@peterbrightwell are heartbeats more consistent to how IS-04 does it now?
Finally does anyone have any other issues or can I merge this in?
